server/neptune/gateway/event/source/project_finder.go

```diff
@@ -0,0 +1,271 @@
+// Copyright 2017 HootSuite Media Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the License);
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//    http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an AS IS BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// Modified hereafter by contributors to runatlantis/atlantis.
+
+package source
+
+import (
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/runatlantis/atlantis/server/core/config/valid"
+
+	"github.com/docker/docker/pkg/fileutils"
+	"github.com/pkg/errors"
+	"github.com/runatlantis/atlantis/server/events/models"
+)
+
+// ProjectFinder determines which projects were modified in a given pull
+// request.
+type ProjectFinder interface {
+	// DetermineProjects returns the list of projects that were modified based on
+	// the modifiedFiles. The list will be de-duplicated.
+	// absRepoDir is the path to the cloned repo on disk.
+	DetermineProjects(modifiedFiles []string, repoFullName string, absRepoDir string, autoplanFileList string) []models.Project
```

Comment:
Do we need this if we're asserting an atlantis.yaml file?

Reply:
Current implementation runs that as a default option should the `atlantis.yaml` file be missing. If we want to enforce its existing, then we can drop it.